Add a methlab basement, new palette for residential basements

anothersimulacrum opened the pull request:
#### Summary
```SUMMARY: Content "Adds basement with a hidden methlab"```


#### Purpose of change
The game was previously lacking a methlab basement. This aims to fix it.


#### Describe the solution
Add a few new itemgroups, a new palette, and a new basement. Two varieties, one with methlab, and one without.


#### Additional context
I might clean up the `methlab` item group and some itemgroup stuff in a few later PR, and I plan on adding some more residential basements.
Pictures:
![LivepeopleLab](https://user-images.githubusercontent.com/42699974/54886128-991a7700-4e41-11e9-95c3-ad05459ed792.png)
![RetrodaysLab](https://user-images.githubusercontent.com/42699974/54886129-99b30d80-4e41-11e9-90cc-9bf76898a161.png)
![ASCIILab](https://user-images.githubusercontent.com/42699974/55017993-2c24ef80-4faf-11e9-818d-2c1e26cc1114.png)
![LivepeopleNoLab](https://user-images.githubusercontent.com/42699974/54886130-99b30d80-4e41-11e9-9ee2-8f974294a579.png)
![RetrodaysNoLab](https://user-images.githubusercontent.com/42699974/54886131-99b30d80-4e41-11e9-9f19-09844e8cbb51.png)
![LivepeopleNoLab2](https://user-images.githubusercontent.com/42699974/55017994-2cbd8600-4faf-11e9-9836-e6183644696b.png)
![RetrodaysNoLab2](https://user-images.githubusercontent.com/42699974/55017995-2cbd8600-4faf-11e9-9fd7-03c4169d6506.png)

Night-Pryanik commented:
Hiding enter to meth lab behind a bookcase makes sense, of course, but most players won't find it, ever. They have absolutely no clue that there might be something behind that bookcase. I encounterd the same problem when I designed my autodoc basement. You should add some hint - I don't know, maybe some field, or graffiti, or chems lying on the bookcase.

Night-Pryanik commented:
That's a really big basement - it has two bedrooms, two bathrooms, a large dining room, a hall and several utility rooms. It looks like a completely functional home rather than a basement.

umsretuza commented:
Initially I thought the screenshots were showcasing the house itself and not the basement. It resembles a surface home/2nd floor quite well. Might want to dial back on the home-like feeling, and more towards an illegal den feeling.

Photoloss commented:
> That's a really big basement - it has two bedrooms, two bathrooms, a large dining room, a hall and several utility rooms. It looks like a completely functional home rather than a basement.

A rare spawn imitating European styles could make sense unless New England laws prohibit it. However the walls should be made of concrete in that case (bonus points for a red brick house on top!) and this motivation does not line up with the intent of the PR: the chance of a Europhile *also* being a DIY methhead becomes vanishingly small.

anothersimulacrum commented:
Updated OP with pictures.
I removed the second bedroom, bathroom, dining room, and rec room, and added some random crate spawns.
Methlab variant will now have meth spawn on the bookshelves, toxic gas in the air and a body/knife/bleach in the laundry room to encourage investigation.

DracoGriffin commented:
Is the toxic gas permanent? That'd be a bit weird.

Also, I thought the previous basement was fine (albeit too large); it's a bit cliche to always assume it wouldn't look clean. It's a bit strange to assume only rednecks or uneducated folk are somehow masterminds of the meth business.

anothersimulacrum commented:
I could add back in the previous basement with a low weight. The toxic gas ins't permanent, which does bring about the problem that it might fade before you find it. The crates are just to imply that it's unfinished, after discussion on the discord.

Mecares commented:
Maybe add a variant that has the same room layout as the Meth Lab but does contain normal rooms, so that the Player might recognise the Basement layout and thinks something like: That is odd normally there is the storage room in this direction instead of those shelves.

anothersimulacrum commented:
Updated, and added more pictures to the OP. Added a new variant with a room behind where the bookshelves go, and added a chance for a hazmat zombie to spawn outside the room in the meth variant.
Also added chance for normal zombies to spawn in all variants, and locked the door to the laundry room in the meth variant, though it's not in the pictures.

ZhilkinSerg commented on data/json/mapgen/basement/basement_meth.json:
```diff
@@ -0,0 +1,232 @@
+        { "monster": "mon_zombie_hazmat", "x": [ 19, 21 ], "y": [ 15, 20 ], "chance": 50 },
+        { "monster": "mon_zombie_hazmat", "x": [ 13, 14 ], "y": [ 11, 16 ], "chance": 50 },
+        { "monster": "mon_zombie_rot", "x": 15, "y": 6, "chance": 30 },
+        { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] }
```
There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?

anothersimulacrum replied:
Yeah, I must have pushed old changes.

ZhilkinSerg commented on data/json/mapgen/basement/basement_meth.json:
```diff
@@ -0,0 +1,232 @@
+        { "group": "home_hw", "x": [ 21, 22 ], "y": 5, "chance": 40, "repeat": [ 1, 3 ] }
+      ],
+      "place_item": [ { "x": 11, "y": 10, "item": "towel" } ],
+      "place_monster": [ { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] } ]
```
There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?

ZhilkinSerg commented on data/json/mapgen/basement/basement_meth.json:
```diff
@@ -0,0 +1,232 @@
+        { "group": "guns_shotgun_rare", "x": 21, "y": 14, "chance": 10 }
+      ],
+      "place_item": [ { "x": 11, "y": 10, "item": "towel" }, { "x": 21, "y": 14, "item": "wrench" } ],
+      "place_monster": [ { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] } ]
```
There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?
